File: typescript/src/api/resources/agents/client/requests/CreateManagedSessionRequest.ts

/**
 * @example
 *     {
 *         environmentId: "environmentId"
 *     }
 */
export interface CreateManagedSessionRequest {
    environmentId: string;
    externalRunId?: string | null;
    externalThreadId?: string | null;
    idempotencyKey?: string | null;
}

File: ruby/lib/Sikaru/agents/types/create_managed_session_request.rb
# frozen_string_literal: true

module Sikaru
  module Agents
    module Types
      class CreateManagedSessionRequest < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :agent_id, -> { String }, optional: false, nullable: false

        field :environment_id, -> { String }, optional: false, nullable: false, api_name: "environmentId"

        field :external_run_id, -> { String }, optional: true, nullable: false, api_name: "externalRunId"

        field :external_thread_id, -> { String }, optional: true, nullable: false, api_name: "externalThreadId"

        field :idempotency_key, -> { String }, optional: true, nullable: false, api_name: "idempotencyKey"
      end
    end
  end
end
